react/SearchWrapper.tsx

```diff
@@ -116,21 +117,61 @@ const getSearchIdentifier = (
   return query + map + (orderBy ?? '') + (page ?? '')
 }
 
-function getCanonicalMaybeWithPage(canonicalLink: string, page: number) {
-  if (page <= 0) {
+interface GetSearchCanonicalParams {
+  canonicalLink: string
+  page?: number
+  map?: string
+}
+
+function shouldNotIncludeMap(map?: string) {
+  if (!map || map === 'b' || map === 'brand') {
+    return true
+  }
+
+  const mapTree = map.split(',')
+
+  if (mapTree.length > 3) {
+    return false
+  }
+
+  return mapTree.every(mapItem => mapItem === 'c')
+}
+
+function getSearchCanonical({
+  canonicalLink,
+  page,
+  map,
+}: GetSearchCanonicalParams) {
+  if (page !== null && page !== undefined && page < 1) {
     return null
   }
 
-  const canonicalWithPage = `${canonicalLink}?page=${page}`
+  const query = {
+    page: page !== undefined && page !== null && page > 1 ? page : undefined,
```

Comment:
I tried to do this, but the lint kept complaining because `undefined > 1` isn't a valid operation, so it could throw an error. This way it's safe to do the comparison.